dl/src/main/scala/com/intel/analytics/bigdl/nn/JoinTable.scala

```diff
@@ -27,6 +27,11 @@ import scala.reflect.ClassTag
  * It is a table module which takes a table of Tensors as input and
  * outputs a Tensor by joining them together along the dimension `dimension`.
  *
+ * Note: the input tensor can have indefinite number of dimensions,
+ * so modules can not automatically inference whether the user use the batch or not
+ * and what the actual dimensions of input excluding the batch dimension.
+ * Hence `nInputDims` represents the actual dimensions of given input excluding the batch.
+ *
```

Comment:
You may want to change the scala doc for the `nInputDims` parameter accordingly too.